File: lib/sm/models/venue.rb
module SM

    class Venue < Model
        belongs_to_tenant

        has_code_identifier from: :name

        validates :name, :capacity, presence: true

        has_one :logo, as: :owner, class_name: 'Hippo::Asset',
                dependent: :destroy, export: true

        def time_in_zone(time)
            ActiveSupport::TimeZone.find_tzinfo(self.timezone).utc_to_local(time)
        end
    end



end

File: client/sm/models/venue.js
import moment from 'moment-timezone';
import Asset from 'hippo/models/asset';
import {
    CachedModel, identifiedBy, identifier, belongsTo, field,
} from './base';

@identifiedBy('sm/venue')
export default class Venue extends CachedModel {

    @identifier id;

    @field code = '';

    @field name = '';

    @field address = '';

    @field phone_number;

    @field capacity;

    @field message_id;

    @field online_sales_halt_mins_before = 30;

    @field timezone = moment.tz.guess();

    @belongsTo({ model: Asset, inverseOf: 'owner' }) logo;

}
